processor/src/lib.rs

```diff
@@ -586,16 +586,20 @@ impl Process {
         match decorator {
             Decorator::Advice(injector) => {
                 let advice_provider = host.advice_provider_mut();
-                let process_state: ProcessState = self.into();
+                let process_state: ProcessState = (&*self).into();
                 match injector {
                     AdviceInjector::MerkleNodeMerge => {
                         advice_provider.merge_merkle_nodes(process_state)?;
                     },
                     AdviceInjector::MerkleNodeToStack => {
                         advice_provider.copy_merkle_node_to_adv_stack(process_state)?
                     },
-                    AdviceInjector::MapValueToStack { include_len, key_offset } => advice_provider
-                        .copy_map_value_to_adv_stack(process_state, *include_len, *key_offset)?,
+                    AdviceInjector::MapValueToStack => {
+                        advice_provider.copy_map_value_to_adv_stack(process_state, false, 0)?
```

Comment:
Are you planning to remove these specialized methods from the advice provider in this or a separate PR?